app/models/journal_cutoff_date.rb

```diff
@@ -0,0 +1,73 @@
+class JournalCutoffDate < ActiveRecord::Base
+  include ActiveModel::ForbiddenAttributesProtection
+
+  validates :cutoff_date, presence: true
+  validate :unique_month_for_cutoff_date, if: :cutoff_date?
+
+  def self.recent_and_upcoming
+    where("cutoff_date > ?", 1.month.ago.beginning_of_month).order(:cutoff_date)
+  end
+
+  def self.past
+    where("cutoff_date < ?", Time.current).order(:cutoff_date)
+  end
+
+  def self.first_valid_date
+    past.last.try(:cutoff_date).try(:beginning_of_month)
+  end
+
+  def last_valid_date
+    cutoff_date.prev_month.end_of_month
+  end
+
+  def past?
+    cutoff_date < Time.current
+  end
+
+  # TODO consider turning this into a reusable module
+  def cutoff_date_time=(hash)
+    formatted = "#{cutoff_date.to_date} #{hash[:hour]}:#{hash[:minute]}#{hash[:ampm]}"
+    self.cutoff_date = Time.zone.parse(formatted)
+  end
```

Comment:
I'm not sure transforming user input belongs in a model.  But it's probably not worth the extra effort to move it somewhere else.